wcfsetup/install/files/lib/form/RegisterForm.class.php

```diff
@@ -429,9 +429,11 @@ public function save() {
 		
 		// generate activation code
 		$addDefaultGroups = true;
-		if (!empty($this->blacklistMatches) || (REGISTER_ACTIVATION_METHOD == 1 && !$registerVia3rdParty) || REGISTER_ACTIVATION_METHOD == 2) {
+		if (!empty($this->blacklistMatches) || (REGISTER_ACTIVATION_METHOD & User::REGISTER_ACTIVATION_USER && !$registerVia3rdParty) || (REGISTER_ACTIVATION_METHOD & User::REGISTER_ACTIVATION_ADMIN)) {
```

Comment:
I wonder if the behavior of `$registerVia3rdParty` should be adjusted to:
- Enable the user.
- But not enable the email address.

This would allow the user to start participating immediately, but still ensures proper double opt in behavior, without needing to rely on a third party.

Reply:
I don't know to what extent that makes sense to completely change the behaviour there. Can't you assume that if you register with a third party, you have a valid email address. Don't third-party registrations only work with validated e-mails anyway?

Reply:
Keep it as it is, we do ~trust~ _not question_ the validity of emails provided by third party services.